### ISSUE TITLE
Unintended war with city state

### ISSUE THREAD

**Issue by ilteroi** [LoneGazebo/Community-Patch-DLL#9084]
often a city state is dragged into a war because they are allied with a major, but during the course of the war the alliance ends (or is replaced by a different major). but the state of war remains. even worse, even when you make peace with the major, you are still at war with the city state!

this is annoying. options?

* city states make peace automatically when the "reason" for the war goes away
* show a notification if there is a city state you can make peace with
* ?

**Comment by LessRekkless** [comment]
The AI seems decent enough at ending wars with City States from de-allied enemies when it wants to do so.

**Comment by ilteroi** [comment]
this is not about AI

**Comment by RecursiveVision** [comment]
City-States making peace automatically when their ally does would be the least disruptive; peace with City-State allies is already included in peace deals. This would just extend it to peace by other means (like third party deals or capitulation) as well.

**Comment by ilteroi** [comment]
there are several cases from the perspective of the minor (X). maybe it makes sense to differentiate. assumption is always there are two majors at war (A and B) and a third major which is neutral (C).

* alliance changes during the war: C -> A. already handled, X joins the war on B.
* alliance changes during the war. A -> C. *not handled*. X stays at war with B. should make peace automatically?
* alliance begins during the war. none -> A. handled. X joins war on B.
* alliance expires during the war. A -> none. *not handled*. X stays at war with B. should make peace automatically?

**Comment by LessRekkless** [comment]
> this is not about AI

No, it's about whether we should let players *choose* to continue their war against said City State without paying another war declaration penalty. The AI can handle it, so it's mostly about whether it's not annoying for a human to have to end their war manually, or being forced into peace against the CD against their will.
